clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx

```diff
@@ -87,43 +67,45 @@ const CreateCustomFields = forwardRef(
       onSubmitComplete();
     };
 
-    const loadList = useCallback((): Map<string, ItemOption> => {
-      const list = new Map<string, ItemOption>();
-      if (data?.length) {
-        data.forEach((value, index) => {
-          if (value?.id && value?.name) {
-            list.set(value.name, { value: value.id });
-            if (index === 0) {
-              initialValues.allow_list_id = value.id;
-            }
-          }
-        });
-      }
-      list.set("Select...", { value: "" });
-      return list;
-    }, [data]);
-
-    const list = loadList();
```

Comment:
[Post-review]: This wasn't memoizing. `useCallback` returns a closure that only changes when the dependency list changes. However, the _return value_ of that function isn't memoized, just the function itself. So this code was still calling the function on every render, doing all of that work, and returning a new list, even when the dependencies hadn't changed.